jupyter_server/services/contents/filemanager.py

```diff
@@ -84,14 +84,33 @@ def _validate_post_save_hook(self, proposal):
             raise TraitError("post_save_hook must be callable")
         return value
 
-    def run_post_save_hook(self, model, os_path):
```

Comment:
Let's put a deprecation cycle on this method name since it's a public API. 

We can shim this method to the new method name, `run_post_save_hooks` and raise a warning in this method. This is likely a low risk API to change, but we should raise a warning just in case.

Reply:
Should we also replace the help message in the `post_save_hook` trait with something like "DEPRECATED, use `register_post_save_hook() instead`"?

Reply:
See my comment below: https://github.com/jupyter-server/jupyter_server/pull/696#discussion_r822184681